charts/agent-control-deployment/templates/_helpers.tpl

```diff
@@ -150,101 +150,62 @@ readOnlyRootFilesystem: true
-{{- if (((.Values.config).fleet_control).auth).organizationId -}}
-  {{- .Values.config.fleet_control.auth.organizationId -}}
-{{- else -}}
-  {{- fail ".config.fleet_control.auth.organizationId is required" -}}
+{{- with (.Values.systemIdentity).organizationId -}}
+  {{- . -}}
 {{- end -}}
 {{- end -}}
 
 
 
 {{- /*
-Check if .Values.config.auth.secret.name exists and use it to name auth' secret. If it does not exist, fallback to the name
-of the releases with "-auth" suffix.
+Check if .Values.systemIdentity.secretName exists and use it to name auth secret. If it does not exist, fallback to the name of the releases with "-auth" suffix.
 */ -}}
 {{- define "newrelic-agent-control.auth.secret.name" -}}
-{{- $secretName := (((((.Values.config).fleet_control).auth).secret).name) -}}
+{{- $secretName := (.Values.systemIdentity).secretName -}}
 {{- if $secretName -}}
   {{- $secretName -}}
 {{- else -}}
   {{- include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "auth" ) }}
 {{- end -}}
 {{- end -}}
 
-
-
 {{- /*
 Helper to toggle the creation of the job that creates and registers the system identity.
 */ -}}
-{{- define "newrelic-agent-control.auth.secret.shouldRunJob" -}}
-{{- $privateKey := include "newrelic-agent-control.auth.secret.privateKey.data" . -}}
-{{- $clientId := include "newrelic-agent-control.auth.secret.clientId.data" . -}}
+{{- define "newrelic-agent-control.check.system.identity.inputs" -}}
 
-{{- if and ((.Values.config).fleet_control).enabled ((((.Values.config).fleet_control).auth).secret).create (not $privateKey) (not $clientId) -}}
-  true
-{{- end -}}
+{{- if and (include "newrelic-agent-control.auth.identityClientSecret" .) (include "newrelic-agent-control.auth.identityClientAuthToken" .) -}}
+  {{- fail "You should not specify in .Values.systemIdentity both a identityClientSecret and a identityClientAuthToken" -}}
 {{- end -}}
 
-
-
-{{- /*
-Helper to toggle the creation of the secret that has the system identity as values.
-*/ -}}
-{{- define "newrelic-agent-control.auth.secret.shouldTemplate" -}}
-{{- if and ((.Values.config).fleet_control).enabled ((((.Values.config).fleet_control).auth).secret).create -}}
-  {{- $privateKey := include "newrelic-agent-control.auth.secret.privateKey.data" . -}}
-  {{- $clientId := include "newrelic-agent-control.auth.secret.clientId.data" . -}}
-
-  {{- if and $privateKey $clientId -}}
-    true
-  {{- else if or $privateKey $clientId -}}
-    {{- fail "If you provide your own system identity data you have to provide both private key and client id" -}}
-  {{- end -}}
-{{- end -}}
+{{- if and (include "newrelic-agent-control.auth.customIdentitySecretName" .) (include "newrelic-agent-control.auth.parentIdentity" .) -}}
+  {{- fail "You should not specify in .Values.systemIdentity both a customSecretName and identityClientId identityClientSecret/identityClientAuthToken" -}}
```

Comment:
I think we also need to remove the `custom` prefix from messages (and maybe the helper names) `customSecretName` --> `secretName` as done in `Values`

Reply:
ups 100% right!